docs/design/interface/pseudocode/check_data.py

```diff
@@ -0,0 +1,64 @@
+# ruff: noqa
+def check_data(
+    data: DataFrame, resource_properties: ResourceProperties
+) -> DataFrame | Error:
```

Comment:
```suggestion
) -> DataFrame:
```
We want to throw an error, not return it, right?

Reply:
Ha, that's right hahahaha